merkle/maphasher/maphasher.go

```diff
@@ -106,7 +106,7 @@ func (m *MapHasher) initNullHashes() {
 	// There are Size()*8 edges, and Size()*8 + 1 nodes in the tree.
 	nodes := m.Size()*8 + 1
 	r := make([][]byte, nodes, nodes)
-	r[0] = m.HashLeaf(0, nil, nil)
+	r[0], _ = m.HashLeaf(0, nil, nil)
```

Comment:
Looks to be dropping an error on the floor here. Please fix or TODO.

Reply:
done

Reply:
+1

I'd suggest returning the error in `initNullHashes` and a panic at `(MapHasher) New` (as systems relying on MapHasher wouldn't really work without it anyway)